From d93819d29d8971187df7b8c770af462d769810de Mon Sep 17 00:00:00 2001
From: Steve Markgraf <steve@steve-m.de>
Date: Tue, 1 May 2018 12:46:31 +0200
Subject: [PATCH 1/3] change version to 0.6git

Signed-off-by: Steve Markgraf <steve@steve-m.de>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9fbda6..d6fb801 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,8 +39,8 @@ endif()
 
 # Set the version information here
 set(VERSION_INFO_MAJOR_VERSION 0) # increment major on api compatibility changes
-set(VERSION_INFO_MINOR_VERSION 5) # increment minor on feature-level changes
-set(VERSION_INFO_PATCH_VERSION 4) # increment patch for bug fixes and docs
+set(VERSION_INFO_MINOR_VERSION 6) # increment minor on feature-level changes
+set(VERSION_INFO_PATCH_VERSION git) # increment patch for bug fixes and docs
 include(Version) # setup version info
 
 ########################################################################
-- 
2.11.0

