From 326b0e3accc21b47e7b4bc118ed4c4b55b914127 Mon Sep 17 00:00:00 2001
From: Martin Hauke <mardnh@gmx.de>
Date: Tue, 1 May 2018 14:04:41 +0200
Subject: [PATCH 2/3] Fix build with MinGW

Signed-off-by: Steve Markgraf <steve@steve-m.de>
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6fb801..abc8f9b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,9 @@ endif()
 ########################################################################
 find_package(PkgConfig)
 find_package(LibUSB)
-set(THREADS_USE_PTHREADS_WIN32 true)
+if(WIN32 AND NOT MINGW)
+    set(THREADS_USE_PTHREADS_WIN32 true)
+endif()
 find_package(Threads)
 
 if(NOT LIBUSB_FOUND)
-- 
2.11.0

