From 7f2ae12548a1a2f74ae2d33e37fec0d141ed5b8f Mon Sep 17 00:00:00 2001
From: Oliver Smith <osmith@sysmocom.de>
Date: Thu, 11 Jul 2019 11:00:06 +0200
Subject: [PATCH 03/33] contrib/jenkins.sh: run "make maintainer-clean"

Related: OS#3047
Signed-off-by: Oliver Smith <osmith@sysmocom.de>
Signed-off-by: Steve Markgraf <steve@steve-m.de>
---
 contrib/jenkins.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/jenkins.sh b/contrib/jenkins.sh
index cb566f1..0538406 100755
--- a/contrib/jenkins.sh
+++ b/contrib/jenkins.sh
@@ -36,5 +36,6 @@ LD_LIBRARY_PATH="$inst/lib" \
   DISTCHECK_CONFIGURE_FLAGS="--enable-werror" \
   $MAKE distcheck \
   || cat-testlogs.sh
+$MAKE maintainer-clean
 
 osmo-clean-workspace.sh
-- 
2.30.2

