From e3e6ee23b7f052327bf64c6908f5c09b75029edc Mon Sep 17 00:00:00 2001
From: Steve Markgraf <steve@steve-m.de>
Date: Sun, 27 Nov 2016 12:19:44 +0100
Subject: [PATCH 12/16] lib: add new HanfTek dongle

Signed-off-by: Steve Markgraf <steve@steve-m.de>
---
 rtl-sdr.rules   | 3 +++
 src/librtlsdr.c | 1 +
 2 files changed, 4 insertions(+)

diff --git a/rtl-sdr.rules b/rtl-sdr.rules
index 4213454..b2f4054 100644
--- a/rtl-sdr.rules
+++ b/rtl-sdr.rules
@@ -72,6 +72,9 @@ SUBSYSTEMS=="usb", ATTRS{idVendor}=="1554", ATTRS{idProduct}=="5020", MODE:="066
 # Astrometa DVB-T/DVB-T2 (R828D)
 SUBSYSTEMS=="usb", ATTRS{idVendor}=="15f4", ATTRS{idProduct}=="0131", MODE:="0666"
 
+# HanfTek DAB+FM+DVB-T
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="15f4", ATTRS{idProduct}=="0133", MODE:="0666"
+
 # Compro Videomate U620F (E4000)
 SUBSYSTEMS=="usb", ATTRS{idVendor}=="185b", ATTRS{idProduct}=="0620", MODE:="0666"
 
diff --git a/src/librtlsdr.c b/src/librtlsdr.c
index cdf1ca9..9b7ba52 100644
--- a/src/librtlsdr.c
+++ b/src/librtlsdr.c
@@ -329,6 +329,7 @@ static rtlsdr_dongle_t known_devices[] = {
 	{ 0x0ccd, 0x00e0, "Terratec NOXON DAB/DAB+ USB dongle (rev 2)" },
 	{ 0x1554, 0x5020, "PixelView PV-DT235U(RN)" },
 	{ 0x15f4, 0x0131, "Astrometa DVB-T/DVB-T2" },
+	{ 0x15f4, 0x0133, "HanfTek DAB+FM+DVB-T" },
 	{ 0x185b, 0x0620, "Compro Videomate U620F"},
 	{ 0x185b, 0x0650, "Compro Videomate U650F"},
 	{ 0x185b, 0x0680, "Compro Videomate U680F"},
-- 
2.11.0

