From fa3a113b77b6ed2738778a52ae4f9ce4e6f18ea3 Mon Sep 17 00:00:00 2001
From: Dimitri Stolnikov <horiz0n@gmx.net>
Date: Mon, 12 Jun 2017 00:04:05 +0200
Subject: [PATCH 15/19] rtl_tcp: add command for bias tee control

---
 src/rtl_tcp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/rtl_tcp.c b/src/rtl_tcp.c
index 317e0f3..aef120e 100644
--- a/src/rtl_tcp.c
+++ b/src/rtl_tcp.c
@@ -355,6 +355,10 @@ static void *command_worker(void *arg)
 			printf("set tuner gain by index %d\n", ntohl(cmd.param));
 			set_gain_by_index(dev, ntohl(cmd.param));
 			break;
+		case 0x0e:
+			printf("set bias tee %d\n", ntohl(cmd.param));
+			rtlsdr_set_bias_tee(dev, (int)ntohl(cmd.param));
+			break;
 		default:
 			break;
 		}
-- 
2.11.0

