From b04c2f9f035c5aede43d731e5d58e4725d2f8bb4 Mon Sep 17 00:00:00 2001
From: Hoernchen <la@tfc-server.de>
Date: Tue, 19 Sep 2017 19:23:00 +0200
Subject: [PATCH 17/19] fix for msvc14

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0597600..c6c6928 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,6 +55,9 @@ if(CMAKE_COMPILER_IS_GNUCC AND NOT WIN32)
     ADD_DEFINITIONS(-Wdeclaration-after-statement)
     #http://gcc.gnu.org/wiki/Visibility
     add_definitions(-fvisibility=hidden)
+elseif(MSVC14 OR MSVC14)
+#pthread-w32 issue, timespec is now part of time.h
+    ADD_DEFINITIONS(-D_TIMESPEC_DEFINED)
 endif()
 
 ########################################################################
-- 
2.11.0

